### PR TITLE
Update iqtree wrapper to 2.1.2

### PR DESCRIPTION
**Pull request by davebx**
Many parameter names were changed, and a few new options were added.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR updates an existing tool or tool collection

### CODE REVIEW

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@davebx if you want you could add `expect_num_outputs` to the test.

@mtekman do you want to look at this one?

File: tools/iqtree/iqtree.xml [hunk 1]
```diff
@@ -313,185 +426,221 @@ Note that -st CODON is always necessary when using codon models and you also nee
...
                         <param argument="-m" type="select" label="Perform standard model selection like jModelTest (for DNA) and ProtTest (for protein)" >
-                            <help><![CDATA[
-<b>Note:</b> Only <code>TEST</code> and custom models that do not rely on <code>MF</code> can be used in conjunction with bootstrap parameters (-b)<br/><br/>
-IQ-TREE also works for codon, binary and morphogical data.
-<table>
-    <tr>
-        <td><i>TESTONLY</i></td>
-        <td>Perform standard model selection like jModelTest (for DNA) and ProtTest (for protein). Moreover, IQ-TREE also works for codon, binary and morphogical data.</td>
-    </tr>
-    <tr>
```
Comment (mtekman) [inline tools/iqtree/iqtree.xml:324]: Some of the help text for the options below are still valid, right? Maybe still good to keep them in

File: tools/iqtree/iqtree.xml [hunk 2]
```diff
@@ -313,185 +426,221 @@ Note that -st CODON is always necessary when using codon models and you also nee
...
-                <param argument="-rbal" type="boolean" truevalue="-rbal" falsevalue="" checked="false" label="Like -r, but a random balanced tree is created."/>
-                <param argument="-rcsg" type="boolean" truevalue="-rcsg" falsevalue="" checked="false" label="Like -r, bur a random circular split network is created."/>
-                <param argument="-rlen" type="text" optional="true" label="Specify three numbers: minimum, mean and maximum branch lengths of the random tree">
-                    <sanitizer>
-                        <valid initial="string.digits,string.whitespace" />
-                    </sanitizer>
+                <param argument="--rand" type="boolean" truevalue="--rand UNI" falsevalue="" checked="false" label="Like -r, but a random tree is created under uniform model."/>
+                <param argument="--rand" type="boolean" truevalue="--rand CAT" falsevalue="" checked="false" label="Like -r, but a random caterpillar tree is created."/>
+                <param argument="--rand" type="boolean" truevalue="--rand BAL" falsevalue="" checked="false" label="Like -r, but a random balanced tree is created."/>
```
Comment (mtekman) [inline tools/iqtree/iqtree.xml:591]: How are these three parameters resolved - are they mutually exclusive? 
In the cheetah, there's only one reference to it: https://github.com/galaxyproject/tools-iuc/pull/3871/files#diff-8a6520fa03f68ddcc80f17cc9fbaafa43c79b3639b286642aaa7ef66bcf83214R294

Is one being "on" then masking all the others which are "off"? If that's true, then ignore this comment :-)

Maybe a select option is better here

**Review by mtekman — COMMENTED** [review]
A fantastic, thorough, much needed upgrade!

I just had some small comments inline above, and also maybe one or two new tests to test out the new features -- like "ancestral_state" "symmetry_test" "conconcordance_factor" (but this is a huge toolset so no need to be super detailed with it)

File: tools/iqtree/iqtree.xml (hunk 2 above)
Comment (davebx) [reply tools/iqtree/iqtree.xml:591]: You're right, and my apologies, I didn't see these comments until I'd already merged, I'll get those changes pushed soon.

Comment (davebx) [reply tools/iqtree/iqtree.xml:591]: I've applied your suggestions in #3872

**Comment by niemasd** [comment]
As per @davebx's request, the following zip contains a multiple sequence alignment and dates that should work in IQ-TREE 2's new date mode:

[sd.sequences_dates.zip](https://github.com/galaxyproject/tools-iuc/files/6970708/sd.sequences_dates.zip)

**Pull request merged**